addons/loyalty/wizard/loyalty_generate_wizard_views.xml

```diff
@@ -33,6 +33,10 @@
                             <field name="valid_until"/>
                         </group>
                     </group>
+                    <div class="alert alert-warning text-center" invisible="customer_ids or mode == 'anonymous'" role="alert">
+                        You are about to generate <span invisible="program_type != 'ewallet'"> an </span><span invisible="program_type != 'gift_card'"> a </span>
+                        <field name="program_type"/> with a value of <field name="points_granted" readonly="1"/> for <field name="coupon_qty" readonly="1"/> customers
```

Comment:
This will be a nightmare for translators, please generate the string cleanly with a dedicated computed field:
```python
@api.depends('program_type', 'points_granted', 'coupon_qty')
def _compute_confirmation_message(self):
    self.confirmation_message = False
    for wizard in self:
        program_desc = dict(wizard._fields['program_type']._description_selection(wizard.env))
        wizard.confirmation_message = _(
            "You're about to generate %(program_type)s with a value of %(value)s for %(customer_number)i customers",
            program_type=program_desc[wizard.program_type],
            value=wizard.points_granted,
            customer_number=wizard.coupon_qty,
       )